node/node.go

```diff
@@ -576,8 +582,10 @@ func (n *nodeImpl) OnStart() error {
 		}
 	}
 
-	if err := n.pexReactor.Start(); err != nil {
-		return err
+	if n.config.P2P.PexReactor {
+		if err := n.pexReactor.Start(); err != nil {
+			return err
+		}
```

Comment:
Is it necessary to check this here?

Reply:
certainly safer. regardless we need to check if it's nil before we start it in any case, and given that `n.pexReactor` is an interface type, the possibility of the field being non-nil while holding a nil implementation exists, and checking the config value is a safer option.

Reply:
can the pex reactor be `nil`?

Reply:
Ah sorry I got confused. I thought this method was just for the seed node but then I realised that both the full node and seed node share the same implementation. Ignore this comment